packages/iov-bns/src/bnsconnection.ts

```diff
@@ -648,24 +647,23 @@ export class BnsConnection implements AtomicSwapConnection {
 
   // updateEscrowBalance will query for the proper balance and then update the accounts of escrow before
   // returning it. Designed to be used in a map chain.
-  protected async updateEscrowBalance<T extends AtomicSwap>(escrow: T): Promise<T> {
-    const addr = conditionToAddress(this.chainId(), escrowCondition(escrow.data.id.data));
+  protected async updateSwapAmounts<T extends AtomicSwap>(swap: T): Promise<T> {
+    const addr = conditionToAddress(this.chainId(), swapCondition(swap.data));
     const account = await this.getAccount({ address: addr });
     const balance = account ? account.balance : [];
-    return { ...escrow, data: { ...escrow.data, amounts: balance } };
+    return { ...swap, data: { ...swap.data, amounts: balance } };
   }
 
   /**
    * Queries the blockchain for the enforced anti-spam fee
    */
   protected async getDefaultFee(): Promise<Amount> {
-    const res = await this.query("/", Encoding.toAscii("gconf:cash:minimal_fee"));
-    if (res.results.length !== 1) {
-      throw new Error("Received unexpected number of fees");
+    const { results } = await this.query("/", Encoding.toAscii("_c:cash"));
+    if (results.length !== 1) {
+      throw new Error(`Unexpected number of results for minimal fee. Expected: 1 Got: ${results.length}`);
     }
-    const data = Encoding.fromAscii(res.results[0].value);
-    const amount = decodeJsonAmount(data);
-    return amount;
+    const { minimalFee } = decodeCashConfiguration(codecImpl.cash.Configuration.decode(results[0].value));
```

Comment:
Yes. This is correct.
We aim to provide proper protobuf objects for all configuration items in a predictable location.
I hoe this works for you, at least better than the previous one, and previously parsed and normalized from the genesis file input/

Reply:
Yes the new protobuf config is way better than what we had before